### PR TITLE
Tidy up the InteractiveEvaluator

### PR DESCRIPTION
**Pull request by amcasey**
Make (and enforce) the simplifying assumption that the `CurrentWindow` property will be set exactly once.  Also tidy up some event un/registration.

### CODE REVIEW

**Comment by amcasey** [comment]
FYI @dotnet/roslyn-interactive

File: src/Interactive/EditorFeatures/Core/InteractiveEditorFeaturesResources.resx [hunk 1]
```diff
@@ -117,6 +117,12 @@
   <resheader name="writer">
     <value>System.Resources.ResXResourceWriter, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
   </resheader>
+  <data name="ReferencesCommandDescription" xml:space="preserve">
```
Comment (amcasey) [inline src/Interactive/EditorFeatures/Core/InteractiveEditorFeaturesResources.resx]: These are actually for a subsequent change, but I suppose there's no harm in merging them now.

**Comment by KevinH-MS** [comment]
:+1:

**Comment by genlu** [comment]
:+1:

**Pull request merged**